scapy/contrib/cansocket_python_can.py

```diff
@@ -91,16 +109,46 @@ def mux(self):
 
 
 class SocketsPool(object):
-    __instance = None
+    """Helper class to organize all SocketWrapper and SocketMapper objects"""
+    __instance = None  # type: Optional[SocketsPool]
 
     def __new__(cls):
+        # type: (Type[object]) -> SocketsPool
+        """Enforces the singleton pattern for SocketsPool class"""
         if SocketsPool.__instance is None:
             SocketsPool.__instance = object.__new__(cls)
-            SocketsPool.__instance.pool = dict()
-            SocketsPool.__instance.pool_mutex = threading.Lock()
+            SocketsPool.__instance._pool = dict()  # type: ignore
+            SocketsPool.__instance._pool_mutex = threading.Lock()  # type: ignore  # noqa: E501
```

Comment:
I wonder if it wouldn't be cleaner to do something like
```
class _SocketsPool(...)
    ...

SocketsPool = _SocketsPool()
```
instead of storing an instance as a class argument. This is super messy

Reply:
I changed it. Thanks for your feedback.